### PR TITLE
Make CI work

### LINKED ISSUE #94: Fix CI
CI is out of date. See https://github.com/jpvillaisaza/slack-web/pull/93#issuecomment-1233516085 for a recommendation to use `haskell-ci` or switch to GitHub Actions.

### PR DESCRIPTION
**Pull request by lf-**
Same PR as #95 but post repo move

This uses GitHub Actions and haskell-ci. I decided not to figure out CircleCI.

You do need the latest haskell-ci version from git to be able to generate for GHC 9.4.2. But it works!

This also builds and tests with GHC 9.4.2 ;-)

It also comes with a bonus Nix flake, because I can't help myself I guess 😆 

Fixes #94

### CODE REVIEW

**Comment by friedbrice** [comment]
Oh, wow! Did we just inherit this package from the old maintainers?

File: cabal.haskell-ci [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  source-repository-package
+    type: git
+    location: https://github.com/nikita-volkov/refined
+    tag: eced2bb0991bde971646e4b3d291870d0aab83a3
+    --sha256: sha256-QvZSFeAmgdBwdyneGRKMCMNGnP+8rD/uTED0icQB+4s=
+
+  allow-newer:
+    -- fixed but not released on hackage yet
+    -- https://github.com/chessai/these-skinny/pull/3
+    these-skinny:base
```
Comment (friedbrice) [inline cabal.haskell-ci]: The these-skinny package on Hackage has 4.18 as its upper bound for base. GHC 9.4 uses base@4.17. We shouldn't need this `allow-newer`, correct?

https://hackage.haskell.org/package/these-skinny-0.7.5

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  nixConfig.allow-import-from-derivation = true; # cabal2nix uses IFD
+
+  outputs = { self, nixpkgs, flake-utils }:
+    let
+      ghcVer = "ghc924";
```
Comment (friedbrice) [inline flake.nix:13]: Don't we want this to be GHC 9.4? Or am I confused about something?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            let haskellPackages = pkgs.haskell.packages.${ghcVer};
+            in
+            haskellPackages.shellFor {
+              packages = p: [ self.packages.${system}.slack-web ];
+              withHoogle = true;
+              buildInputs = with haskellPackages; [
+                haskell-language-server
+                fourmolu
+                # jacked on this particular nixpkgs version
+                # ghcid
```
Comment (friedbrice) [inline flake.nix:58]: Oh, no! D-:

**Review by friedbrice — COMMENTED** [review]
Looks great! Thanks for adopting this package.

I had some questions, as there are a few things I don't understand.

File: cabal.haskell-ci (hunk 1 above)
Comment (lf-) [reply cabal.haskell-ci]: sick, they released it :D

File: flake.nix (hunk 2 above)
Comment (lf-) [reply flake.nix:13]: This is for dev purposes; HLS kinda sucks on 9.4 at present. I just used the flake template I had, but more versions could also be added in the future.

Comment (lf-) [reply flake.nix:13]: Also 9.4 is not yet fixed up in nixpkgs, which w1ggles is working on fixing https://github.com/NixOS/nixpkgs/pull/191618 :)

File: flake.nix (hunk 3 above)
Comment (lf-) [reply flake.nix:58]: I'm going to run a bunch of compile and we'll see if a newer nixpkgs version fixes it. Unfortunately there's no caching for ghc924 on macOS/aarch64.

Comment (lf-) [reply flake.nix:58]: hahahaha lol, fourmolu is jacked on new nixpkgs. I know why, however.

**Review by friedbrice — APPROVED** [review]

**Pull request merged**